app/src/main/java/de/westnordost/streetcomplete/quests/surface/AddCyclewayPartSurface.kt

```diff
@@ -0,0 +1,29 @@
+package de.westnordost.streetcomplete.quests.surface
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.meta.OsmTaggings
+import de.westnordost.streetcomplete.data.osm.SimpleOverpassQuestType
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder
+import de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao
+
+class AddCyclewayPartSurface(o: OverpassMapDataDao) : SimpleOverpassQuestType<String>(o) {
+
+    override val tagFilters = """
+        ways with (
+        (bicycle ~ designated|yes and segregated=yes and highway ~ path|footway|cycleway|bridleway and surface = paved)
```

Comment:
Only for `surface = paved`? It should rather be surface = anything paved. I think there is a constant for that in `OsmTaggings`.
Though, maybe drop this altogether, because if it is already tagged as segregated, there is not the danger that the app asks for each and every path in a forest or something.

Reply:
I would expect thing tagged for example `highway=path surface=paving_stones segregated=yes bicycle=designated foot=designated` to have paving stones both for cycleway and footway part.

(of course, some may be tagged incorrectly but...)

Reply:
Why would you expect that?

Reply:
Because using `surface=paving_stones` for `highway=path` representing something like

![300px-Witosa_bike_2](https://user-images.githubusercontent.com/899988/61900357-c07f6e80-af1d-11e9-8f99-2cd94e1e4a13.jpg)

would be incorrect - half of surface is an asphalt.

Assuming correct tagging `surface=paving_stones` would be used for cases where both pedestrian and cyclist part has paving stones surface.

(at least that is how I always tagged such objects and never expected that others would tag it differently - why `surface=paving_stones` would be used in case where half of surface is asphalt?)

Reply:
Hmm, I understand your point. Though, what about `!surface or surface=paved` then?

Reply:
And in turn `surface=paved` is correct where both sides are paved - but exact surface is different.

Reply:
> Though, what about !surface or surface=paved then?

Ones without surface should be caught by generic path surface quest. I hope that people will answer correctly with `surface=paved` in cases like this, and specific surface where it is the same for both.

Reply:
Ok

Reply:
> I hope that people will answer correctly with surface=paved in cases like this

I doubt that though.

I was about to write this in a new comment, when I discovered this here:

---

Only if  surface = paved? Two issues:
* What if it is a more specific value like `paving_stones` or `sett`? (a user could "accidentially" set this for segregated ways only loooking at the footway)
* What if it is not set? (If I get the geneal "paved" quest and I have a segregated way, I would not select anything as a user in a segregated way, because it is segregated. Also SC image for "paved" may bias the user's selection in the general case, because IIRC it shows an asphalt way - maybe it could explicitly show such a segregated way to prevent this?)

IMHO this query is a little too specific.

**Proposed solution:** Why not do it the other way around: exclude such segregated ways in the generic surface quest and _instead_ show this (much more useful/correct/specific) one?

Reply:
This means though that StreetComplete tags the surface of segregated cycle- and sidewalks even if it turns out to be the exact same surface. Is this acceptable?

Reply:
At least it very explicitly confirms it is the same surface. I know this would be a totally new UI, but if the two quests were asked together (and you only have a "left side"/"right side" selection for "is paved"/"is not paved") or so, then you could provide an answer "both have the same surface"...

Reply:
> This means though that StreetComplete tags the surface of segregated cycle- and sidewalks even if it turns out to be the exact same surface. Is this acceptable?

What worse, it leaves `surface` tag unfilled. I think that it is preferable to have `surface=paved cycleway:surface=asphalt footway:surface=paving_stones` over `cycleway:surface=asphalt footway:surface=paving_stones`.

Reply:
It probably is preferable. But the surface-quest is still asked for segregated ways, so the `surface` key will be added sooner or later anyway right?

Reply:
Yes, it is asked. But

> exclude such segregated ways in the generic surface quest and instead show this (much more useful/correct/specific) one?

proposed in this thread would change it, that is why I am skeptical.

Reply:
If it is changed like I proposed earlier - do not ask for the surface of a segregated way - then it would be quite incompatible with the quest filter here.

Reply:
So there are following ways to handle it

(1)
standard path quests asks about all paths, cycleway/footway quest only for segregated with surface=paved and with pedestrian/cyclist access
what may go wrong: user may incorrectly answer with overly specific surface rather than surface=paved

(2)
standard path quests asks about unsegregated paths, cycleway/footway quest for all segregated with pedestrian/cyclist access
what will go wrong: surface tag will be never set, cycleway:surface and footway:surface will be sometimes set to the same values

(3)
standard path quests asks about all ways, cycleway/footway quest for all segregated with pedestrian/cyclist access
what will go wrong: cycleway:surface and footway:surface will be sometimes set to the same values

I still think that (1) is the preferable solution.

Reply:
It is considered okay to tag building:levels on the building outline even if the building contains building:parts. For data consumers which do not process such detail. I find it reasonable to act the same for surfaces